include/psa/crypto_struct.h

```diff
@@ -287,12 +287,28 @@ static inline void psa_make_key_persistent(psa_key_attributes_t *attributes,
     attributes->lifetime = lifetime;
 }
 
+static inline void psa_set_key_id(psa_key_attributes_t *attributes,
+                                  psa_key_id_t id)
+{
+    attributes->id = id;
+    if( attributes->lifetime == PSA_KEY_LIFETIME_VOLATILE )
+        attributes->lifetime = PSA_KEY_LIFETIME_PERSISTENT;
+}
+
 static inline psa_key_id_t psa_get_key_id(
     const psa_key_attributes_t *attributes)
 {
     return( attributes->id );
 }
 
+static inline void psa_set_key_lifetime(psa_key_attributes_t *attributes,
+                                        psa_key_lifetime_t lifetime)
+{
+    attributes->lifetime = lifetime;
+    if( lifetime == PSA_KEY_LIFETIME_VOLATILE )
+        attributes->id = 0;
```

Comment:
So, 0 is now officially an invalid id? What's the best place to document this?

Reply:
Rework of https://github.com/ARMmbed/mbed-crypto/pull/104 coming up.